test/https-gateway.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is the shared Gateway for all Knative routes to use.
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: knative-ingress-gateway
+  namespace: knative-serving
+  labels:
+    serving.knative.dev/release: devel
+    networking.knative.dev/ingress-provider: istio
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    hosts:
+    - "*"
+    tls:
+      mode: PASSTHROUGH
```

Comment:
`PASSTHOUGH` actually does not work. Can we change it to:
```
tls:
      mode: SIMPLE
      credentialName: "tls-certificate"
    hosts:
    - "*"
```
We can think about how to generate the certificate as a TODO.

Reply:
@ZhiminXiang I just applied this config and tried running the TestHelloWorld e2e test, but looking at the logs, routes were ready but they were not https. The test failed with. Am I missing something here? 
```
    kubelogs.go:152: I 13:03:08.438 [route-controller] [serving-tests/hello-world-uyuibzpu] Updating targeted revisions.
    kubelogs.go:152: I 13:03:08.438 [route-controller] [serving-tests/hello-world-uyuibzpu] Creating placeholder k8s services
    kubelogs.go:152: I 13:03:08.455 [route-controller] [serving-tests/hello-world-uyuibzpu] Creating Ingress.
    kubelogs.go:152: I 13:03:08.482 [route-controller] [serving-tests/hello-world-uyuibzpu] Updating placeholder k8s services with clusterIngress information
    kubelogs.go:152: I 13:03:08.482 [route-controller] [serving-tests/hello-world-uyuibzpu] Route successfully synced
    kubelogs.go:152: I 13:03:08.482 [route-controller] [serving-tests/hello-world-uyuibzpu] Reconcile succeeded. Time taken: 44.180415ms.
    service.go:133: Checking to ensure Service Status is populated for Ready service hello-world-uyuibzpu
    service.go:139: Getting latest objects Created by Service hello-world-uyuibzpu
    kubelogs.go:152: D 13:03:09.241 [kpa-class-podautoscaler-controller] [serving-tests/hello-world-uyuibzpu-qm8r9] No data to scale on yet
    service.go:142: Successfully created Service hello-world-uyuibzpu
    helloworld_test.go:66: The endpoint for Route hello-world-uyuibzpu at domain hello-world-uyuibzpu.serving-tests.169.46.25.10.xip.io didn't serve the expected text "Hello World! How about some tasty noodles?": response: status: 404, body: , headers: map[Content-Length:[0] Date:[Mon, 19 Aug 2019 13:03:09 GMT] Server:[istio-envoy] Zipkin_trace_id:[34de09e4cccf986c15324e8dc95c5bc9]] did not pass checks: status = 404, want one of: [200]
```

Reply:
Actually I missed setting up the CA issuer (thanks @rmoe for the instructions). The TestHelloWorld test passes now